siliconcompiler/core.py

```diff
@@ -3855,11 +3857,25 @@ def _setup_tool(self, tool, task, step, index):
         self.set('arg','step', step)
         self.set('arg','index', index)
 
-        func = self.find_function(tool, 'setup', 'tools')
-        if func is None:
+        # Generic tool setup.
+        setup_tool = self.find_function(tool, 'setup', 'tools')
+        if setup_tool:
+            setup_tool(self)
```

Comment:
Remove `setup_tool` function: call common setup from task `setup()`, and error out if no task `setup()` exists.